### PR TITLE
Added koko eating bananas problem (binary search on answer)

### PR DESCRIPTION
**Pull request by Atreya-112**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.

### CODE REVIEW

File: src/main/java/com/thealgorithms/binarysearchonanswer/BinarySearchOnAnswer.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+     */
+
+
+    private static boolean blackBox(int piles[],int h,int k){
+        for(int i = 0;i < piles.length;i++){
+            h -=  (int)Math.ceil((double)piles[i]/(double)k); 
+        }
+        return (h >= 0);
+    }
+}
```
Comment (vil02) [inline src/main/java/com/thealgorithms/binarysearchonanswer/BinarySearchOnAnswer.java:106]:
```suggestion
}

```
This should fix the failing `clang-format` job.

**Review by vil02 — CHANGES_REQUESTED** [review]
Please do not use the `main` method. I think in this case, `BinarySearchOnAnswer` can be a _utility class_. And please add unit tests of your class. I think a good examples to look at are [`Factorial`](https://github.com/TheAlgorithms/Java/blob/34cf6dab28ee2271fbf228d345bcf14db85c211c/src/main/java/com/thealgorithms/maths/Factorial.java) and [`FactorialTest`](https://github.com/TheAlgorithms/Java/blob/34cf6dab28ee2271fbf228d345bcf14db85c211c/src/test/java/com/thealgorithms/maths/FactorialTest.java).

File: src/main/java/com/thealgorithms/binarysearchonanswer/BinarySearchOnAnswer.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+ If the pile has less than k bananas, 
+ she eats all of them instead and will not eat any more bananas during this hour.
+ Koko likes to eat slowly but still wants to finish eating all the bananas before the guards return.
+ Return the minimum integer k such that she can eat all the bananas within h hours.
+ 
+ example test cases : 
+ Input: piles = [3,6,7,11], h = 8
+ Output: 4
+
+ leetcode link : https://leetcode.com/problems/koko-eating-bananas/descripti
```
Comment (BamaCharanChhandogi) [inline src/main/java/com/thealgorithms/binarysearchonanswer/BinarySearchOnAnswer.java:21]: check leetcode link

**Comment by siriak** [comment]
We do not add leetcode problems, as they are usually just a variation of a known technique. Those basic techniques are already implemented separately

**Pull request closed without merging**